src/v/storage/spill_key_index.cc

```diff
@@ -109,15 +109,38 @@ spill_key_index::spill_some(size_t entry_size, size_t min_index_size) {
      * Collect keys to spill in a buffer until the stop condition is met and
      * then dump all of them at once into the backing file. A periodic yield is
      * introduced to reduce the chance of a reactor stall.
+     *
+     * In order to avoid creating hot erase spots we start removal at the
+     * location immediately after the last inserted key. The exact last inserted
+     * key is skipped to maintain good hit locality, and the next key should not
```

Comment:
> The exact last inserted key is skipped to maintain good hit locality

This read a bit weird to me looking through this again. To clarify, what I mean is that we shouldn't remove the last inserted key for fear of destroying workloads that repeat the same key.